src/app/api/utils/auth.ts

```diff
@@ -201,8 +193,26 @@ async function fetchUserInfo(accessToken: string) {
       Authorization: `Bearer ${accessToken ?? ""}`,
     },
   });
-  const userInfo = (await response.json()) as Profile;
-  return userInfo;
+  const userInfo = await response.json();
+  return userInfo as Profile;
```

Comment:
Actually, come to think of it, isn't this type assertion inaccurate? Which would explain why TypeScript didn't complain. (And relatedly, shouldn't line 82 still pass the return value through `convertFxaProfile`?)

```suggestion
  return userInfo as FxaProfile;
```

Reply:
Oh weird, good catch, hm... login is failing with this change though, there must be something wrong with the conversion function. I'm surprised it worked before.

Reply:
(next-auth is complaining that `id` is missing when I clearly see it on the object)

Reply:
Here's what it did before:
```
-        request: async (context) => {
-          const response = await fetch(AppConstants.OAUTH_PROFILE_URI, {
-            headers: {
-              Authorization: `Bearer ${context.tokens.access_token ?? ""}`,
-            },
-          });
-          const userInfo = (await response.json()) as Profile;
-          return userInfo;
-        },
```

I think what is happening here is that the next-auth `UserinfoEndpointHandler` that we're passing the `fetchUserInfo` function to ends up calling `profile`, which is where this `FxAProfile` - `Profile` conversion used to live.

I'll see if I can re-implement in a way that makes this clearer...

Reply:
Hmm, am I missing something? It looks like you have changed this now, and it still works for me? :)